PyAngle/AngleTest.py

```diff
@@ -89,35 +89,34 @@ def test_cmp(self):
         self.assertTrue(angle1 == angle5)
 
     def test_trigonometric_functions(self):
-        angle1 = Angle(degree=30)
-        self.assertAlmostEqual(angle1.sin(), 0.50, delta=0.01)
-        self.assertAlmostEqual(angle1.cos(), 0.87, delta=0.01)
-        self.assertAlmostEqual(angle1.tan(), 0.57, delta=0.01)
+        angle1 = Angle.from_dms(deg=30)
+        self.assertAlmostEqual(0.50, angle1.sin(), delta=0.01)
+        self.assertAlmostEqual(0.87, angle1.cos(), delta=0.01)
+        self.assertAlmostEqual(0.57, angle1.tan(), delta=0.01)
 
     def test_toXY(self):
-        angle1 = Angle(degree=30)
+        angle1 = Angle.from_dms(deg=30)
         self.assertTupleEqual(
-            angle1.toXY(), (math.cos(math.radians(30)), math.sin(math.radians(30)))
+            (math.cos(math.radians(30)), math.sin(math.radians(30))), angle1.to_atan2()
         )
-        self.assertTupleEqual(angle1.toXY(x=1), (1, math.tan(math.radians(30))))
-        self.assertTupleEqual(angle1.toXY(y=math.sqrt(3)), (3, math.sqrt(3)))
+        self.assertTupleEqual((1, math.tan(math.radians(30))), angle1.to_atan2(x=1))
+        self.assertTupleEqual((3, math.sqrt(3)), angle1.to_atan2(y=math.sqrt(3)))
 
     def test_judge_angle_types(self):
-        self.assertTrue(Angle(degree=0).isZeroAngle())
-        self.assertTrue(Angle(degree=30).isAcuteAngle())
-        self.assertTrue(Angle(degree=30).isMinorAngle())
-        self.assertFalse(Angle(degree=30).isRightAngle())
-        self.assertTrue(Angle(degree=90).isRightAngle())
-        self.assertTrue(Angle(degree=100).isObtuseAngle())
-        self.assertFalse(Angle(degree=100).isRightAngle())
-        self.assertFalse(Angle(degree=100).isAcuteAngle())
-        self.assertFalse(Angle(degree=100).isStraightAngle())
-        self.assertTrue(Angle(degree=180).isStraightAngle())
-        self.assertFalse(Angle(degree=190).isMinorAngle())
-        self.assertTrue(Angle(degree=110).isMinorAngle())
-        self.assertTrue(Angle(degree=200).isMajorAngle())
+        self.assertTrue(Angle.from_dms(deg=0).isZeroAngle())
+        self.assertTrue(Angle.from_dms(deg=30).isAcuteAngle())
+        self.assertTrue(Angle.from_dms(deg=30).isMinorAngle())
+        self.assertFalse(Angle.from_dms(deg=30).isRightAngle())
+        self.assertTrue(Angle.from_dms(deg=90).isRightAngle())
+        self.assertTrue(Angle.from_dms(deg=100).isObtuseAngle())
+        self.assertFalse(Angle.from_dms(deg=100).isRightAngle())
+        self.assertFalse(Angle.from_dms(deg=100).isAcuteAngle())
+        self.assertFalse(Angle.from_dms(deg=100).isStraightAngle())
+        self.assertTrue(Angle.from_dms(deg=180).isStraightAngle())
+        self.assertFalse(Angle.from_dms(deg=190).isMinorAngle())
+        self.assertTrue(Angle.from_dms(deg=110).isMinorAngle())
+        self.assertTrue(Angle.from_dms(deg=200).isMajorAngle())
 
 
 if __name__ == "__main__":
-    unittest.main()
-
+    unittest.main()
```

Comment:
Don't DALAO checked my silly test cases? (dog head

Reply:
You can handle test cases well. I trust you. I'm not DALAO, I'm a noob.